app/controllers/admin/application_pets_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Admin::ApplicationPetsController < ApplicationController
+
+  def update
+    application = Application.find(params[:application_id])
+    pet = Pet.find(params[:pet_id])
+    match = ApplicationPet.find_by(pet_id: pet.id, application_id: application.id)
+    if params[:approve] == "true"
+      match.approve
+      pet.approve
```

Comment:
Does a pet need approving in addition to the ApplicationPet joins record status? What does pet approval do?

Reply:
I wonder if you could reduce the complexity here by having a general `update_status` method in the PetApplication model and you could pass the `params[:approve]` in as a parameter. That way you could push the complexity down into a model method and keep the controller a bit slimmer. This isn't an MVC violation but just a design idea!

Reply:
I see now that pet.approve is changing the adoptable flag! My bad